src/pages/Staking/index.tsx

```diff
@@ -1,10 +1,747 @@
+  } = useQuery<MonetaryAmount<Currency<VoteUnit>, VoteUnit>, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'votingBalance',
+      address
+    ],
+    genericFetcher<MonetaryAmount<Currency<VoteUnit>, VoteUnit>>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(voteGovernanceTokenBalanceError);
+
+  // My Rewards
+  const {
+    isIdle: rewardAmountAndAPYIdle,
+    isLoading: rewardAmountAndAPYLoading,
+    data: rewardAmountAndAPY,
+    error: rewardAmountAndAPYError,
+    refetch: rewardAmountAndAPYRefetch
+  } = useQuery<RewardAmountAndAPY, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getRewardEstimate',
+      address
+    ],
+    genericFetcher<RewardAmountAndAPY>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(rewardAmountAndAPYError);
+
+  // Estimated KINT Rewards & APY
+  const monetaryLockingAmount = newMonetaryAmount(lockingAmount, GOVERNANCE_TOKEN, true);
+  const {
+    isIdle: estimatedRewardAmountAndAPYIdle,
+    isLoading: estimatedRewardAmountAndAPYLoading,
+    data: estimatedRewardAmountAndAPY,
+    error: estimatedRewardAmountAndAPYError
+  } = useQuery<RewardAmountAndAPY, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getRewardEstimate',
+      address,
+      monetaryLockingAmount
+    ],
+    genericFetcher<RewardAmountAndAPY>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(estimatedRewardAmountAndAPYError);
+
+  const {
+    isIdle: stakedAmountAndEndBlockIdle,
+    isLoading: stakedAmountAndEndBlockLoading,
+    data: stakedAmountAndEndBlock,
+    error: stakedAmountAndEndBlockError,
+    refetch: stakedAmountAndEndBlockRefetch
+  } = useQuery<StakedAmountAndEndBlock, Error>(
+    [
+      GENERIC_FETCHER,
+      'interBtcApi',
+      'escrow',
+      'getStakedBalance',
+      address
+    ],
+    genericFetcher<StakedAmountAndEndBlock>(),
+    {
+      enabled: !!bridgeLoaded
+    }
+  );
+  useErrorHandler(stakedAmountAndEndBlockError);
+
+  const initialStakeMutation = useMutation<void, Error, LockingAmountAndTime>(
+    (variables: LockingAmountAndTime) => {
+      if (currentBlockNumber === undefined) {
+        throw new Error('Something went wrong!');
+      }
+      const unlockHeight = currentBlockNumber + getLockBlocks(variables.time);
+
+      return window.bridge.interBtcApi.escrow.createLock(variables.amount, unlockHeight);
+    },
+    {
+      onSuccess: () => {
+        voteGovernanceTokenBalanceRefetch();
+        stakedAmountAndEndBlockRefetch();
+        rewardAmountAndAPYRefetch();
+        reset({
+          [LOCKING_AMOUNT]: '0.0',
+          [LOCK_TIME]: '0'
+        });
+      }
+    }
+  );
+
+  const moreStakeMutation = useMutation<void, Error, LockingAmountAndTime>(
+    (variables: LockingAmountAndTime) => {
+      return (async () => {
+        if (currentBlockNumber === undefined) {
```

Comment:
When extending the lock time, it is actually extended from the `endBlock` from the `getStakedBalance` call.